jcl/src/java.base/share/classes/java/lang/invoke/MethodHandleHelper.java

```diff
@@ -0,0 +1,528 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package java.lang.invoke;
+
+import java.lang.invoke.MethodHandles.Lookup;
+import java.util.ArrayList;
+import java.util.Objects;
+
+import com.ibm.oti.util.Msg;
+import com.ibm.oti.vm.VM;
+import com.ibm.oti.vm.VMLangAccess;
+
+/*[IF Sidecar19-SE]
+import jdk.internal.misc.Unsafe;
+import jdk.internal.reflect.ConstantPool;
+/*[ELSE]*/
+import sun.misc.Unsafe;
+import sun.reflect.ConstantPool;
+/*[ENDIF]*/
+
+import com.ibm.jit.JITHelpers;
+
+/**
+ * Static methods for the MethodHandle class.
+ */
+public final class MethodHandleHelper {
```

Comment:
```suggestion
final class MethodHandleHelper {
```
Needs to be package protected.

This PR is already tested. Shall I update this PR or do this change in a follow-up PR?

Reply:
in this PR please.  We'll rerun tests as required - likely only a single platform

Reply:
updated.